tests/settings_tests/tests.py

```diff
@@ -441,3 +441,132 @@ def test_tuple_settings(self):
             finally:
                 del sys.modules['fake_settings_module']
                 delattr(settings_module, setting)
+
+
+class SettingChangeEnterException(Exception):
+    pass
+
+
+class SettingChangeExitException(Exception):
+    pass
+
+
+class OverrideSettingsIsolationOnExceptionTest(SimpleTestCase):
+    """
+    Test that ``override_settings`` context manager restores settings
```

Comment:
Docstrings should state the expected behavior and omit prefixes like "Tests that" since all tests test things. This is from [Python coding style](https://docs.djangoproject.com/en/dev/internals/contributing/writing-code/coding-style/#python-style).